From fc58cd950cc0666e97012256f6df73fa88b71513 Mon Sep 17 00:00:00 2001
From: Xulin Sun <xulin.sun@windriver.com>
Date: Wed, 17 Mar 2021 14:11:16 +0800
Subject: [PATCH 2/2] server: remove prot argument for vm_map_ram call

The kernel commit d4efd79a81abc70 ("mm: remove the prot argument from vm_map_ram")
removed prot argument from vm_map_ram call. Here to adapt this kernel
API.

Upstream-Status: Inappropriate [upstream is still using k5.4 version]

Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 services/server/env/linux/osfunc_arm64.c        | 1 +
 services/server/env/linux/physmem_osmem_linux.c | 6 +++---
 services/system/common/env/linux/dma_support.c  | 2 +-
 3 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/services/server/env/linux/osfunc_arm64.c b/services/server/env/linux/osfunc_arm64.c
index da0f4f4..8165544 100644
--- a/services/server/env/linux/osfunc_arm64.c
+++ b/services/server/env/linux/osfunc_arm64.c
@@ -43,6 +43,7 @@ CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 #include <linux/version.h>
 #include <linux/cpumask.h>
 #include <linux/dma-mapping.h>
+#include <linux/dma-map-ops.h>
 #include <asm/cacheflush.h>
 
 #include "pvrsrv_error.h"
diff --git a/services/server/env/linux/physmem_osmem_linux.c b/services/server/env/linux/physmem_osmem_linux.c
index e4bebb4..6d352c4 100644
--- a/services/server/env/linux/physmem_osmem_linux.c
+++ b/services/server/env/linux/physmem_osmem_linux.c
@@ -951,7 +951,7 @@ _ZeroPageArray(IMG_UINT32 uiNumToClean,
 #if !defined(CONFIG_64BIT) || defined(PVRSRV_FORCE_SLOWER_VMAP_ON_64BIT_BUILDS)
 		pvAddr = vmap(ppsCleanArray, uiToClean, VM_WRITE, pgprot);
 #else
-		pvAddr = vm_map_ram(ppsCleanArray, uiToClean, -1, pgprot);
+		pvAddr = vm_map_ram(ppsCleanArray, uiToClean, -1);
 #endif
 		if (!pvAddr)
 		{
@@ -1503,7 +1503,7 @@ _ApplyCacheMaintenance(PVRSRV_DEVICE_NODE *psDevNode,
 			IMG_CPU_PHYADDR sUnused =
 				{ IMG_CAST_TO_CPUPHYADDR_UINT(0xCAFEF00DDEADBEEFULL) };
 
-			pvAddr = vm_map_ram(ppsCleanArray, uiToClean, -1, pgprot);
+			pvAddr = vm_map_ram(ppsCleanArray, uiToClean, -1);
 			if (!pvAddr)
 			{
 				PVR_DPF((PVR_DBG_ERROR,
@@ -3180,7 +3180,7 @@ PMRAcquireKernelMappingDataOSMem(PMR_IMPL_PRIVDATA pvPriv,
 #if !defined(CONFIG_64BIT) || defined(PVRSRV_FORCE_SLOWER_VMAP_ON_64BIT_BUILDS)
 	pvAddress = vmap(pagearray, ui32PageCount, VM_READ | VM_WRITE, prot);
 #else
-	pvAddress = vm_map_ram(pagearray, ui32PageCount, -1, prot);
+	pvAddress = vm_map_ram(pagearray, ui32PageCount, -1);
 #endif
 	if (pvAddress == NULL)
 	{
diff --git a/services/system/rogue/common/env/linux/dma_support.c b/services/system/rogue/common/env/linux/dma_support.c
index fd111dc..f89a38d 100644
--- a/services/system/rogue/common/env/linux/dma_support.c
+++ b/services/system/rogue/common/env/linux/dma_support.c
@@ -140,7 +140,7 @@ SysDmaAcquireKernelAddress(struct page *psPage, IMG_UINT64 ui64Size, void *pvOSD
 #if !defined(CONFIG_64BIT) || defined(PVRSRV_FORCE_SLOWER_VMAP_ON_64BIT_BUILDS)
 	pvVirtAddr = vmap(pagearray, ui32PgCount, VM_READ | VM_WRITE, prot);
 #else
-	pvVirtAddr = vm_map_ram(pagearray, ui32PgCount, -1, prot);
+	pvVirtAddr = vm_map_ram(pagearray, ui32PgCount, -1);
 #endif
 
 	/* Clean-up tmp buffers */
-- 
2.17.1

